src/client/dfs/dfs.c

```diff
@@ -1849,7 +1838,47 @@ dfs_lookup(dfs_t *dfs, const char *path, int flags, dfs_obj_t **_obj,
 	for (token = strtok_r(rem, "/", &sptr);
 	     token != NULL;
 	     token = strtok_r(NULL, "/", &sptr)) {
-dfs_lookup_loop:
+lookup_rel_path_loop:
+
+		/*
+		 * Open the directory object one level up.
+		 * Since fetch_entry does not support ".",
+		 * we can't support ".." as the last entry,
+		 * nor can we support "../.." because we don't
+		 * have parent.parent_oid and parent.mode.
+		 * For now, represent this partial state with
+		 * parent_fully_valid.
+		 */
+		parent_fully_valid = true;
+		if (strcmp(token, "..") == 0) {
+			parent_fully_valid = false;
+
+			/* Cannot go outside the container */
+			if (daos_oid_cmp(parent.oid, dfs->root.oid) == 0) {
+				D_GOTO(err_obj, rc = ENOENT);
```

Comment:
not required, but printing a debug message here might be helpful

Reply:
I agree. I added one